changelog.d/pr-8557

```diff
@@ -0,0 +1,20 @@
+synopsis: Add `--semaphore` flag to enable interaction with GHC Job Server protocol
+packages: cabal-install
+prs: #8557
+
+description: {
+
+When cabal-install is passed the `--semaphore` flag it will now act as a job server
+according to the GHC Jobserver Protocol.
+
+In particular this means that cabal-install will create a semaphore which it then
+passes to `./Setup build` (and hence `ghc`) which can be used by `ghc` in order to
+control how much parellism it uses in coordation with other simultaneously running
+processes.
+
+This feature requires ghc-9.8 in order to use as this is the first version of GHC
+which implements the protocol.
+
+The protocol is specified by [GHC Proposal #123]().
```

Comment:
```suggestion
The protocol is specified by [GHC Proposal #540]().
```

The proposal number should be 540, but it hasn't been merged yet although it's been accepted.